crates/bevy_render/src/renderer/render_resource/render_resource_bindings.rs

```diff
@@ -228,12 +235,53 @@ impl RenderResourceBindings {
             })
     }
 
-    fn build_bind_group(&self, bind_group_descriptor: &BindGroupDescriptor) -> Option<BindGroup> {
+    fn build_bind_group(
+        &mut self,
+        bind_group_descriptor: &BindGroupDescriptor,
+        render_resources: &dyn RenderResourceContext,
+    ) -> Option<BindGroup> {
         let mut bind_group_builder = BindGroup::build();
         for binding_descriptor in bind_group_descriptor.bindings.iter() {
             if let Some(binding) = self.get(&binding_descriptor.name) {
+                // If the default texture view is not correct, we must correct it.
+                let binding = if let BindType::Texture {
+                    // This case only happens in the Cube and CubeArray cases,
+                    // since their underlying textures are all 2d texture arrays.
+                    view_dimension:
+                        view_dimension @ (TextureViewDimension::Cube | TextureViewDimension::CubeArray),
+                    ..
+                } = binding_descriptor.bind_type
+                {
+                    if let RenderResourceBinding::Texture(original_view) = binding {
+                        // This will return a cached one for this particular
+                        // bindgroup.
+                        // TODO: Add support for custom texture views somehow?
+                        let new_view = render_resources.create_texture_view(
+                            original_view.get_texture_id(),
+                            TextureViewDescriptor {
+                                dimension: Some(view_dimension),
+                                ..Default::default()
+                            },
+                            Some(bind_group_descriptor.id),
+                        );
+                        if new_view != *original_view {
+                            self.set(
+                                &binding_descriptor.name,
+                                RenderResourceBinding::Texture(new_view),
+                            );
+                            RenderResourceBinding::Texture(new_view)
+                        } else {
+                            binding.clone()
+                        }
+                    } else {
+                        binding.clone()
+                    }
+                } else {
+                    binding.clone()
+                };
```

Comment:
There's probably a way to simplify this `if`/`else` chain so that we don't have to write `binding.clone()` on each `else` branch.

Reply:
I managed to get it down to two by collapsing the two `if let`s into one, but I can't think of any way which will play nicely with 
the current borrowck.

Reply:
you may have used a nightly feature for that 😄

Reply:
Yes... I should probably have switched to stable when running tests and compiling 😅